corehq/warehouse/models/staging.py

```diff
@@ -8,46 +8,52 @@
 from corehq.apps.users.models import CouchUser
 from corehq.apps.groups.models import Group
 from corehq.apps.domain.models import Domain
+from casexml.apps.phone.models import SyncLog
+from corehq.form_processor.models import XFormInstanceSQL
 from corehq.warehouse.dbaccessors import (
     get_group_ids_by_last_modified,
     get_user_ids_by_last_modified,
     get_domain_ids_by_last_modified,
+    get_synclog_ids_by_date,
+    get_forms_by_last_modified,
 )
-from corehq.warehouse.utils import django_batch_records
+from corehq.warehouse.const import (
+    GROUP_STAGING_SLUG,
+    USER_STAGING_SLUG,
+    DOMAIN_STAGING_SLUG,
+    FORM_STAGING_SLUG,
+    SYNCLOG_STAGING_SLUG,
+)
+
+from corehq.warehouse.models.shared import WarehouseTableMixin
+from corehq.warehouse.etl import CouchToDjangoETLMixin
 
 
-class StagingTable(models.Model):
+class StagingTable(models.Model, WarehouseTableMixin):
 
     class Meta:
         abstract = True
 
     @classmethod
-    def raw_record_iter(cls, start_datetime, end_datetime):
-        raise NotImplementedError
-
-    @classmethod
-    def field_mapping(cls):
-        # Map source model fields to staging table fields
-        # ( <source field>, <staging field> )
-        raise NotImplementedError
-
-    @classmethod
     @transaction.atomic
```

Comment:
Is it necessary to do this as an atomic operation? Considering it's size it would be quite expensive.